File: python/bob/bob.py
def hey(phrase):
    """
    Given statements, returns the answers of Bob,
    a lackadaisical teenager.
    """
    if phrase.strip() == '':
        return "Fine. Be that way!"
    if phrase.isupper():
        return "Whoa, chill out!"
    if phrase.strip().endswith("?"):
        return "Sure."
    return "Whatever."

File: typescript/bob/bob.ts
class Bob {
  private isQuiet(dialogue: string): boolean {
    return dialogue === ``
  }

  private isYelling(dialogue: string): boolean {
    return (
      dialogue === dialogue.toUpperCase() &&
      dialogue.toUpperCase() !== dialogue.toLowerCase()
    )
  }

  private isQuestion(dialogue: string): boolean {
    return dialogue.endsWith(`?`)
  }

  hey(dialogue: string): string {
    const trimmed: string = dialogue.trim()
    if (this.isQuiet(trimmed)) {
      return `Fine. Be that way!`
    }
    if (this.isYelling(trimmed)) {
      return `Whoa, chill out!`
    }
    if (this.isQuestion(trimmed)) {
      return `Sure.`
    }
    return `Whatever.`
  }
}

export default Bob
